### PR TITLE
Add actuator endpoint to find and delete sessions

### PR DESCRIPTION
**Pull request by eddumelendez**
- [X] I have signed the CLA

See gh-407

### CODE REVIEW

File: spring-session-boot/src/main/java/org/springframework/session/endpoint/SessionEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * {@link MvcEndpoint} to expose actuator session.
+ *
+ * @author Eddú Meléndez
+ * @since 1.2.0
+ */
```
Comment (tsachev) [inline spring-session-boot/src/main/java/org/springframework/session/endpoint/SessionEndpoint.java]: Why not use `@ConfigurationProperties` here and make `sensible` and `path` configurable?

File: spring-session-boot/src/main/java/org/springframework/session/endpoint/EndpointConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for session management
+ * {@link Endpoint}s.
+ *
+ * @author Eddú Meléndez
+ * @since 1.2.0
+ */
+@Configuration
+@ConditionalOnBean(FindByIndexNameSessionRepository.class)
```
Comment (tsachev) [inline spring-session-boot/src/main/java/org/springframework/session/endpoint/EndpointConfiguration.java]: This cloud be spring boot autoconfiguration registered in `spring.factories`

**Comment by vpavic** [comment]
@eddumelendez This is a very useful addition however IMO it should be targeted at Spring Boot since first-class Spring Session integration already exists there.

WDYT @rwinch?

**Comment by rwinch** [comment]
@vpavic @eddumelendez I do think it may be better fit for Spring Boot as this is where most actuator endpoints live.

**Comment by vpavic** [comment]
@rwinch Also I don't think it's a wise idea to have a dependency on Boot (other than in sample apps that is).

Since Boot's POM actually provides the dependency management for Session that would effectively create a circular dependency between the projects.

**Comment by cemo** [comment]
+1 @vpavic

**Comment by eddumelendez** [comment]
Thanks guys for the feedback I will move my changes to spring-boot project... Hope to see all new improvements there soon.

**Comment by eddumelendez** [comment]
Close in favor of https://github.com/spring-projects/spring-boot/pull/5730

**Pull request closed without merging**